### PR TITLE
Adding IPv6 support to peering

### PR DESCRIPTION
**Pull request by kph7**
Adding IPv6 support to the p2p layer.  Tested between two IPv6 nodes.

### CODE REVIEW

File: p2p/listener.go [hunk 1]
```diff
@@ -159,6 +161,15 @@ func (l *DefaultListener) ExternalAddress() *NetAddress {
 	return l.extAddr
 }
 
+func (l *DefaultListener) ExternalAddressToString() string {
```
Comment (alexanderbez) [inline p2p/listener.go]: Thoughts on adding a godoc for this guy?

**Comment by mdyring** [comment]
Thanks for the work on this.

Is there a way to advertise both ipv4 and ipv6 addresses via external_address (config.toml)?

**Comment by mdyring** [comment]
I can confirm this also worksfor RPC:

```
# netstat -l
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp6       0      0 [::]:26656              [::]:*                  LISTEN
tcp6       0      0 [::]:26657              [::]:*                  LISTEN
```

**Pull request merged**